### PR TITLE
add a job to build foreman-discovery-image on ci.centos.org

### PR DESCRIPTION
**Pull request by evgeni**
this does add a build job, but does not yet connect it to the release job on ci.tfm.o

### CODE REVIEW

File: puppet/modules/jenkins_job_builder/files/centos.org/pipelines/foreman-discovery-image-build.groovy [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+                runPlaybook(
+                    playbook: 'playbooks/setup_forklift.yml',
+                    inventory: cico_inventory('./'),
+                )
+            }
+        }
+        stage('Run Build') {
+            steps {
+                script {
+                    duffy_ssh("git clone https://github.com/${env.repoowner}/foreman-discovery-image/ --branch ${env.branch}", 'duffy_box', './')
```
Comment (ekohl) [inline puppet/modules/jenkins_job_builder/files/centos.org/pipelines/foreman-discovery-image-build.groovy]: Do we ever want to build it not from the official source in our infra? Normally I'm all for configuration but in this case it's probably not desired (unless we go all the way and make it `env.git_url`).

File: puppet/modules/jenkins_job_builder/files/centos.org/pipelines/foreman-discovery-image-build.groovy [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+                    duffy_scp('foreman-discovery-image/aux/vagrant-build/result/', '.', 'duffy_box', './')
+                }
+            }
+        }
+    }
+
+    post {
+        success {
+            archiveArtifacts artifacts: 'result/*tar', allowEmptyArchive: true
+            archiveArtifacts artifacts: 'result/*iso', allowEmptyArchive: true
```
[the remark this answers is not in the document]
Comment (ekohl) [reply puppet/modules/jenkins_job_builder/files/centos.org/pipelines/foreman-discovery-image-build.groovy]: I think that's what https://github.com/theforeman/foreman-infra/pull/1128 is for.

[the remark this answers is not in the document]
Comment (evgeni) [reply puppet/modules/jenkins_job_builder/files/centos.org/pipelines/foreman-discovery-image-build.groovy]: Yepp, I just want to build the image on ci.centos.org and then make ci.tfm.o push that to web02.

**Comment by lzap** [comment]
Thank you very much. Please delete the old job from our CI and set up some process how can I trigger build. If you can make this part of our standard release pipeline/workflow I'd be appreciated. My suggestion would be: build the latest stable git tag against latest stable CentOS7 for each release.

File: puppet/modules/jenkins_job_builder/files/centos.org/pipelines/foreman-discovery-image-build.groovy (hunk 1 above)
Comment (evgeni) [reply puppet/modules/jenkins_job_builder/files/centos.org/pipelines/foreman-discovery-image-build.groovy]: dropped the repoowner param after talking with lzap

**Pull request merged**